airflow/serialization/serde.py

```diff
@@ -288,7 +287,26 @@ def _convert(old: dict) -> dict:
 
 
 def _match(classname: str) -> bool:
-    return any(p.match(classname) is not None for p in _get_patterns())
+    """Checks if the given classname matches a path pattern
+    either using glob format or regexp format
+    """
+    return _match_glob(classname) or _match_regexp(classname)
+
+
+def _match_glob(classname: str):
+    """Checks if the given classname matches a pattern from
+    allowed_deserialization_classes using glob syntax
+    """
+    patterns = conf.get("core", "allowed_deserialization_classes").split()
```

Comment:
Can we improve performance here? Potentially this is called very often so I'd prefer a cache here rather than a look up every time.